### PR TITLE
Added Keda scaler lib

### PR DESCRIPTION
**Pull request by YuriiBudnyi**

### CODE REVIEW

File: kubernetes/config-skeleton.libsonnet [hunk 1]
```diff
@@ -281,5 +281,24 @@
...
     },
+    kedaScalerConfig: {
+      name: 'scaler-%s' % s.deployment.name,
+      trigerConfigs: [],
+      behavior: s.common.autoscaling.behavior,
+      fallback: {
+        failureThreshold: 3,
+        replicas: 2,
+      },
+      scaleTarget: {
```
Comment (PatTheSilent) [inline kubernetes/config-skeleton.libsonnet:292]: This should be configurable. We scale both statefulsets and deployments

File: kubernetes/keda.libsonnet [hunk 2]
```diff
@@ -0,0 +1,51 @@
+local metric_definition(type, queue='', value='', activationThreshold='0', server) =
```
Comment (PatTheSilent) [inline kubernetes/keda.libsonnet]: please be consistent with the function name casing. We're using camel case pretty much everywhere throughout this library. Don't randomly switch naming to snake case.

**Review by PatTheSilent — CHANGES_REQUESTED** [review]

File: kubernetes/kubernetes.libsonnet [hunk 3]
```diff
@@ -354,6 +358,8 @@ local letsbuildServiceDeployment(
       + pdb.spec.selector.withMatchLabels(dc.labels)
   ),
 
+  keda: if withKedaScaler then keda.scaledObject(ks),
```
Comment (PatTheSilent) [inline kubernetes/kubernetes.libsonnet:361]: Don't we want to disable HPA creation if we create a scaledobject?

File: kubernetes/config-skeleton.libsonnet (hunk 1 above)
Comment (YuriiBudnyi) [reply kubernetes/config-skeleton.libsonnet:292]: Done

File: kubernetes/keda.libsonnet (hunk 2 above)
Comment (YuriiBudnyi) [reply kubernetes/keda.libsonnet]: Done

File: kubernetes/kubernetes.libsonnet (hunk 3 above)
Comment (YuriiBudnyi) [reply kubernetes/kubernetes.libsonnet]: Dissabled

**Review by PatTheSilent — APPROVED** [review]

**Pull request merged**